assemblies/assembly-planner-java.adoc

```diff
@@ -0,0 +1,750 @@
+                .filter(lesson -> lesson.getTimeslot() == null || lesson.getRoom() == null)
+                .collect(Collectors.toList());
+        if (!unassignedLessons.isEmpty()) {
+            LOGGER.info("");
+            LOGGER.info("Unassigned lessons");
+            for (Lesson lesson : unassignedLessons) {
+                LOGGER.info("  " + lesson.getSubject() + " - " + lesson.getTeacher() + " - " + lesson.getStudentGroup());
+            }
+        }
+    }
+
+}
+----
+
+
+. Run the `TimeTableApp` class as the main class of a normal Java application. The following output should result:
++
+----
+...
+INFO  |            | Room A     | Room B     | Room C     |
+INFO  |------------|------------|------------|------------|
+INFO  | MON 08:30  | English    | Math       |            |
+INFO  |            | I. Jones   | A. Turing  |            |
+INFO  |            | 9th grade  | 10th grade |            |
+INFO  |------------|------------|------------|------------|
+INFO  | MON 09:30  | History    | Physics    |            |
+INFO  |            | I. Jones   | M. Curie   |            |
+INFO  |            | 9th grade  | 10th grade |            |
+...
+----
+
+. Verify the console output. Does it conform to all hard constraints?
+What happens if you comment out the `roomConflict` constraint in `TimeTableConstraintProvider`?
+
+The `info` log shows what OptaPlanner did in those five seconds:
+
+[source,options="nowrap"]
+----
+... Solving started: time spent (33), best score (-8init/0hard/0soft), environment mode (REPRODUCIBLE), random (JDK with seed 0).
+... Construction Heuristic phase (0) ended: time spent (73), best score (0hard/0soft), score calculation speed (459/sec), step total (4).
+... Local Search phase (1) ended: time spent (5000), best score (0hard/0soft), score calculation speed (28949/sec), step total (28398).
+... Solving ended: time spent (5000), best score (0hard/0soft), score calculation speed (28524/sec), phase total (2), environment mode (REPRODUCIBLE).
+----
+
+== Test the application
+A good application includes test coverage. Test the constraints and the solver in your timetable project.
+
+include::{enterprise-dir}/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/optimizer/business-optimizer-test-solver-quarkus-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/optimizer/business-optimizer-logging-quarkus-proc.adoc[leveloffset=+1]
+
+== Using Micrometer and Prometheus to monitor your school timetable OptaPlanner Java application
+
+OptaPlanner exposes metrics through https://micrometer.io/[Micrometer], a metrics instrumentation library for Java applications. You can use Micrometer with Prometheus to monitor the OptaPlanner solver in the school timetable application.
+
+To configuring a plain Java OptaPlanner application to use Micrometer, you must
+
+
+.Prerequisites
+* You have created the OptaPlanner school timetable application with Java.
```

Comment:
The same as here https://github.com/emmurphy1/kie-docs/pull/336/files#r710880591

Reply:
OK. I think this is covered in the previous comment.

Reply:
LGTM